pkg/authorization/api/validation/validation.go

```diff
@@ -147,16 +151,30 @@ func ValidatePolicy(policy *authorizationapi.Policy, isNamespaced bool) field.Er
 			allErrs = append(allErrs, field.Invalid(keyPath.Child("metadata", "name"), role.Name, "must be "+roleKey))
 		}
 
-		allErrs = append(allErrs, validateRole(role, isNamespaced, keyPath)...)
+		if isCreate {
+			allErrs = append(allErrs, validateRole(role, isNamespaced, keyPath)...) // policy creation validation is more strict
+		}
 	}
 
 	return allErrs
 }
 
 func ValidatePolicyUpdate(policy *authorizationapi.Policy, oldPolicy *authorizationapi.Policy, isNamespaced bool) field.ErrorList {
-	allErrs := ValidatePolicy(policy, isNamespaced)
+	allErrs := validatePolicy(policy, isNamespaced, false)
 	allErrs = append(allErrs, validation.ValidateObjectMetaUpdate(&policy.ObjectMeta, &oldPolicy.ObjectMeta, field.NewPath("metadata"))...)
-
+	rolePath := field.NewPath("roles")
+	for roleKey, role := range policy.Roles {
+		if role == nil {
+			continue // these cause errors in validatePolicy so we do not worry about them
+		}
+		keyPath := rolePath.Key(roleKey)
+		oldRole, isExistingRole := oldPolicy.Roles[roleKey]
+		if isExistingRole {
+			allErrs = append(allErrs, validateRoleUpdate(role, oldRole, isNamespaced, keyPath)...)
```

Comment:
A side-effect of this change is that during a policy update the existing internal roles must have their `ResourceVersion` field set or the whole policy will fail validation (this was not required previously).  It is technically possible to have a policy that was created directly with roles that have no RV set.